tests/ops/api/v1/endpoints/test_email_endpoints.py

```diff
@@ -0,0 +1,497 @@
+        db: Session,
+        api_client: TestClient,
+        payload,
+        url,
+        generate_auth_header,
+    ):
+        auth_header = generate_auth_header([EMAIL_CREATE_OR_UPDATE])
+        response = api_client.post(url, headers=auth_header, json=payload)
+
+        assert 200 == response.status_code
+        response_body = json.loads(response.text)
+
+        assert response_body["key"] == "mailgun"
+        email_config = db.query(EmailConfig).filter_by(key="mailgun")[0]
+        email_config.delete(db)
+
+    def test_post_email_config_with_invalid_key(
+        self,
+        db: Session,
+        api_client: TestClient,
+        payload,
+        url,
+        generate_auth_header,
+    ):
+        payload["key"] = "*invalid-key"
+        auth_header = generate_auth_header([EMAIL_CREATE_OR_UPDATE])
+        response = api_client.post(url, headers=auth_header, json=payload)
+        assert 422 == response.status_code
+        assert (
+            json.loads(response.text)["detail"][0]["msg"]
+            == "FidesKey must only contain alphanumeric characters, '.', '_' or '-'."
+        )
+
+    def test_post_email_config_with_key(
+        self,
+        db: Session,
+        api_client: TestClient,
+        payload,
+        url,
+        generate_auth_header,
+    ):
+        payload["key"] = "my_email_config"
+        auth_header = generate_auth_header([EMAIL_CREATE_OR_UPDATE])
+
+        response = api_client.post(url, headers=auth_header, json=payload)
+        assert 200 == response.status_code
+
+        response_body = json.loads(response.text)
+        email_config = db.query(EmailConfig).filter_by(key="my_email_config")[0]
+
+        expected_response = {
+            "key": "my_email_config",
+            "name": "mailgun",
+            "service_type": EmailServiceType.MAILGUN.value,
+            "details": {
+                EmailServiceDetails.API_VERSION.value: "v3",
+                EmailServiceDetails.DOMAIN.value: "my.mailgun.domain",
+                EmailServiceDetails.IS_EU_DOMAIN.value: False,
+            },
+        }
+        assert expected_response == response_body
+        email_config.delete(db)
+
+    def test_post_email_config_missing_detail(
+        self,
+        api_client: TestClient,
+        url,
+        generate_auth_header,
+    ):
+        auth_header = generate_auth_header([EMAIL_CREATE_OR_UPDATE])
+        response = api_client.post(
+            url,
+            headers=auth_header,
+            json={
+                "key": "my_email_config",
+                "name": "mailgun",
+                "service_type": EmailServiceType.MAILGUN.value,
+                "details": {
+                    # "domain": "my.mailgun.domain"
+                },
+            },
+        )
+        assert response.status_code == 422
+        errors = response.json()["detail"]
+        assert "details" in errors[0]["loc"]
+        assert errors[0]["msg"] == "[\"field required ('domain',)\"]"
```

Comment:
Same with this response, if you get rid of your custom validation in EmailConfigRequest.validate_details, you get the following response body, without the escaping issues:
```json
{
   "loc":[
      "body",
      "details",
      "domain"
   ],
   "msg":"field required",
   "type":"value_error.missing"
}
```